pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -884,6 +865,47 @@ public SuccessResponse deleteSegments(
     }
   }
 
+  @DELETE
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/segments/{tableName}/select")
```

Comment:
Can you please elaborate on why a new api is needed instead of adding more parameters for the existing segment deletion api?

Reply:
We want to be able to delete data older than a given age using an API. At present I think we only have APIs that either delete all segments or delete one particular segment.

This functionality should be useful to everyone since this is a common requirement when you have a lot of customers using Pinot.

Reply:
There's an delete api to delete list of segments, but it's not user friendly. Users need to use the get api with time window and then invoke the deletion api. Moreover, the new delete api flip the default value of excludeOverlapping (false -> true) and excludeReplacedSegments (true -> false) of get api to cater with more intuitive deletion requirement. If the segment min-max partially overlap with the give time window, the segments might contain data out of given time window. Most of customers don't expect to delete those data. Thus excludeOverlapping is set to true, meaning ignore segment that are partially overlapped for deletion. For excludeReplacedSegments, the default value is set to false, meaning completely delete segments regardless of the replaced states.

@jtao15 do you have any objections?